lib/ExpensiMark.js

```diff
@@ -287,7 +287,7 @@ export default class ExpensiMark {
         this.htmlToTextRules = [
             {
                 name: 'breakline',
-                regex: /((<br[^>]*>)+)/gi,
+                regex: /<br[^>]*>/gi,
```

Comment:
I just roll back to the issue and the PR. Back then the issue is about the last message text in the LHN when sending a blockquote. So the solution I propose is to replace the blockquote closing tag with a space. Other regex rules in the `htmlToText` rule from the E/App that I move to expensi-mark.

I believe this will affect the text in LHN where it should be replacing adjoin break lines with a space.

Reply:
> I believe this will affect the text in LHN where it should be replacing adjoin break lines with a space.

I had this thought after I stopped working hehe, but I just checked and it doesn't affect in web:

![image](https://user-images.githubusercontent.com/87341702/203389223-9a01c9b1-b09d-4079-b618-7fc53305af76.png)

I'm going to check Android/IOS

Reply:
Also, I noticed that the server:
- for the report action, it replaces each `<br>` for a `' '` (whitespace), but
     ![image](https://user-images.githubusercontent.com/87341702/203390228-8d435cb9-d908-4780-93aa-52bba13d52dc.png)
- for the report, the `lastMessageText` has collapsed consecutive `<br>` into a single `' '`
     ![image](https://user-images.githubusercontent.com/87341702/203390233-38be8fb1-2db8-4546-81e9-cc0fa4a900f1.png)

Reply:
I guess we should separate the function to convert the message HTML to text between LHN and chat.

Reply:
Yeah, something like that... I was looking at the backend, and convert the `html` to `lastMessageText` using a function that applies the same regexes that `htmlToText` apply, including `/((<br[^>]*>)+)/`.

I was thinking that it doesn't make much sense that `🥶 <br /><br /><br />🥶` gets converted to `🥶  🥶` just collapsing the `<br />` into a single space... why the result is not `🥶 🥶` (single space) if our intention is really to collapse multiple white spaces. I'm thinking that `htmlToText` should maybe be changed to:
- Step 1: replace each `<br />` by single spaces
- Step 2: collapse all consecutive spaces into a single space

I still have to look at what the back end is doing to produce the `text` of the message in the report action from the `html`. Is there a reason why we don't collapse the spaces when deriving `text` from `html`?

Reply:
The `html` of a report action's message gets converted to `text` [here](https://github.com/Expensify/Web-Expensify/blob/7e7c1eb933d5688fbd457211ca104b89feaa6f9b/lib/ReportUtils.php#L1691), and it replaces each `<br />` with a single whitespace (no collapsing).

The `lastMessageText` of the report gets derived from `html` [here](https://github.com/Expensify/Web-Expensify/blob/7e7c1eb933d5688fbd457211ca104b89feaa6f9b/lib/ReportUtils.php#L2285-L2288), collapsing multiple `<br />` into a single whitespace.

Reply:
> I guess we should separate the function to convert the message HTML to text between LHN and chat.

@mollfpr so.. I think the best way to go here is:

- `htmlToText` won't collapse multiple `<br />` into a single whitespace because that is only needed for the `lastMessageText` which is used in the LHN (what this PR is doing)
- The function [ReportUtils.formatReportLastMessageText](https://github.com/Expensify/App/blob/00d7c619cc409b96e6dddca5cc0ad406f0785331/src/libs/ReportUtils.js#L396-L398) will have the responsibility of collapsing consecutive whitespaces into a single one. We already use this function to process what we put in `lastMessageText`

Reply:
@aldo-expensify  sounds good to me!

I found an issue when testing this PR, but it's not related to the PR changes. The last message on LHN flinched without whitespace when sending a text with a blockquote.

https://user-images.githubusercontent.com/25520267/204142869-a6763984-fcb3-4433-90e5-0583b0532ddd.mov

Reply:
Logged issue here https://expensify.slack.com/archives/C049HHMV9SM/p1669564594408699